### PR TITLE
Update recipe for PKG-4520_codesign

### PR DESCRIPTION
**Pull request by cbouss**
Rebuild for PKG-4513 authenticode signing

### CODE REVIEW

**Review by Jrice1317 — APPROVED** [review]
LGTM, but does `cryptography_vectors` need to be rebuilt for code signing as well?

**Comment by danpetry** [comment]
Are the osx sdk and rust compiler versions tied together somehow? [This](https://cloud.prefect.io/anaconda/flow-run/4ba891fa-ccb4-4265-9e28-4f56a54d34dc) previous run succeeded with rust v1.71.1 and sdk v10.10.

**Comment by danpetry** [comment]
the rust compiler update since then seems to have broken it

**Comment by Callek** [comment]
> LGTM, but does `cryptography_vectors` need to be rebuilt for code signing as well?

My evaluation on signing needs was based on what is installed with miniconda. We probably do need to rebuild other things, like that as well. But I created the list which included this based ONLY on miniconda itself

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,9 @@
 # rust compilier specified so the rust-gnu compiler will not be used.
 rust_compiler:
   - rust
+rust_compiler_version:
+  - 1.76.0
+MACOSX_DEPLOYMENT_TARGET:   # [osx and x86_64]
+  - "10.12"                 # [osx and x86_64]
+CONDA_BUILD_SYSROOT:        # [osx and x86_64]
+  - /opt/MacOSX10.12.sdk    # [osx and x86_64]
```
Comment (skupr-anaconda) [inline recipe/conda_build_config.yaml:9]: As Daniel said it's not really needed, right?

Comment (cbouss) [reply recipe/conda_build_config.yaml]: It is to build with 1.76. I added a commit to build with 1.71.

**Pull request merged**